l3experimental/l3bitset/l3bitset.dtx

```diff
@@ -0,0 +1,637 @@
+% Evaluates the \meta{integer expression} and unsets the bit of the
+% index position it represents (sets it to 0).
+% The index is 1-based. If the \meta{integer expression} evaluates to a
+% negative number or is larger than the current length of the bit vector
+% nothing happens. If the leading (left most) bit is unset,
+% zeros are not trimmed but stay in the bit vector.
+% \end{function}
+%
+% \begin{function}[added = 2020-12-22]
+%   {\bitset_clear:N,\bitset_clear:c,\bitset_gclear:N,\bitset_gclear:c}
+%   \begin{syntax}
+%     \cs{bitset_clear:N}  \meta{bitset var} \\
+%     \cs{bitset_gclear:N}  \meta{bitset var}
+%   \end{syntax}
+% This resets the bitset to the initial state.
+% \end{function}
+%
+% \section{Using bitsets}
+%
+% \begin{function}[EXP,added = 2020-12-14]
+%   { \bitset_item:Nn, \bitset_item:cn }
+%   \begin{syntax}
+%     \cs{bitset_item:Nn}   \meta{bitset var}  \Arg{integer expression}
+%   \end{syntax}
+% Evaluates the \meta{integer expression} and output \texttt{1} if the bit with
+% this index number is set and \texttt{0} otherwise.
+% \end{function}
+%
+% \begin{function}[EXP,added = 2020-12-13]
+%   {\bitset_to_bin:N, \bitset_to_bin:c}
+%   \begin{syntax}
+%     \cs{bitset_to_bin:N} \meta{bitset var}
+%   \end{syntax}
+% This leaves the current value of the bitset expressed as
+% a binary (string) number in the input stream.
+% If no bit has been set yet, the output is zero.
+% \end{function}
+% \begin{function}[EXP,added = 2020-12-13]
+%   {\bitset_to_arabic:N, \bitset_to_arabic:c}
+%   \begin{syntax}
+%     \cs{bitset_to_arabic:N} \meta{bitset var}
+%   \end{syntax}
+% This leaves the current value of the bitset expressed as
+% a decimal number in the input stream. If no bit has been set yet,
+% the output is zero. The function uses \cs{int_from_bin:n} if the largest
+% index that have been set or unset is smaller then 32, and a slower implementation
+% based on \cs{fp_eval} otherwise.
+% \end{function}
+%
+%
+% \begin{function}[added = 2020-12-13]
+%   {\bitset_show:N, \bitset_show:c}
+%   \begin{syntax}
+%     \cs{bitset_show:N} \meta{bitset var}
+%   \end{syntax}
+% Displays the binary and decimal value of the \meta{bitset var} on the terminal,
+% \end{function}
+%
+% \begin{function}[added = 2020-12-13]
+%   {\bitset_log:N, \bitset_log:c}
+%   \begin{syntax}
+%     \cs{bitset_log:N} \meta{bitset var}
+%   \end{syntax}
+% Writes the value of the \meta{bitset var} in the log file.
+% \end{function}
+%
+
+% \section{Syntactic sugar: Setting bits by keys}
+%
+% \begin{function}[added = 2020-12-19]
+%   {\bitset_addto_named_index:Nn}
+%   \begin{syntax}
+%     \cs{bitset_addto_named_index:Nn} \meta{bitset var}
+%      \{
+%         \meta{name1} |=| \meta{index1} |,|
+%         \meta{name2} |=| \meta{index2} |,| \ldots{}
+%      \}
+%   \end{syntax}
+% This adds the name--index pairs given in the second
+% argument to an internal, global property list. If the
+% property list didn't exist yet, it is created.
```

Comment:
The fact that this is implemented as a property list seems irrelvant for documentation. By all means it could be done as csnames or what have you. Important on that level is only that this attaches a name to the index position and that this correspondance is global.

Reply:
ok. I will reword it.